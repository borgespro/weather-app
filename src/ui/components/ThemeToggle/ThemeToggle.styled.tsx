import styled from 'styled-components/native';

import { CustomThemeProps } from '@/typings/theme';

export const ThemeToggleContainer = styled.View<CustomThemeProps>`
  width: 100%;
  flex-direction: row;
  justify-content: space-around;
  padding-horizontal: ${(props) => props.theme.spacing.xxl};
  padding-vertical: ${(props) => props.theme.spacing.xl};
`;
