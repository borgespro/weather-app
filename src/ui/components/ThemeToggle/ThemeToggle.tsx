import React from 'react';

import { useThemeSetupContext } from '@/contexts/ThemeSetupContext';

import Button from '@/ui/elements/Button';

import { ThemeToggleContainer } from './ThemeToggle.styled';

export default function ThemeToggle(): React.ReactElement {
  const { currentThemeType, toggleCurrentTheme } = useThemeSetupContext();

  return (
    <ThemeToggleContainer>
      <Button label="Light" onPress={toggleCurrentTheme} active={currentThemeType === 'light'} />
      <Button label="Dark" onPress={toggleCurrentTheme} active={currentThemeType === 'dark'} />
    </ThemeToggleContainer>
  );
}
