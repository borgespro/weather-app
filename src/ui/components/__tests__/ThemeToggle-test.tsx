import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ThemeToggle from '../ThemeToggle';

describe('ThemeToggle Element', () => {
  it('should render without issues', () => {
    const component = shallow(<ThemeToggle />);
    expect(component.length).toBe(1);
    expect(component.children().length).toBe(2);
    expect(toJson(component)).toMatchSnapshot();
  });
});
