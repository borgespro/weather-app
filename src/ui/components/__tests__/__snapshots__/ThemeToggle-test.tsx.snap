// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle Element should render without issues 1`] = `
<Styled(View)>
  <Button
    active={false}
    label="Light"
    onPress={[Function]}
  />
  <Button
    active={true}
    label="Dark"
    onPress={[Function]}
  />
</Styled(View)>
`;
